### ISSUE TITLE
Agate type inference is too aggressive

### ISSUE THREAD

**Issue by drewbanin** [fishtown-analytics/dbt#861]
Via: http://agate.readthedocs.io/en/1.6.0/api/data_types.html

All:
```
Actual:
  null_values = ('', 'na', 'n/a', 'none', 'null', '.')
Desired:
  null_values = ('null')
```

Boolean:
```
Actual:
  true_values = ('yes', 'y', 'true', 't', '1')
  false_values = ('no', 'n', 'false', 'f', '0')
  null_values = ('', 'na', 'n/a', 'none', 'null', '.')
Desired:
  true_values = ('true')
  false_values = ('false')
  null_values = ('null')
```

TimeDelta
```
Actual:
  null_values = ('', 'na', 'n/a', 'none', 'null', '.')
Desired:
  null_values = ('null')
```

This would be a breaking change, so I think it's something we ought to do sooner rather than later.

**Comment by cmcarthur** [comment]
document the heck out of it

**Comment by beckjake** [comment]
fixed in #905
